src/main/java/opendcs/dao/TsGroupDAO.java

```diff
@@ -122,14 +123,12 @@ public TsGroup getTsGroupById(DbKey groupId, boolean forceDbRead)
 //else debug2("READING FROM DATABASE group id " + groupId);
 		
 		String q = "SELECT " + GroupAttributes + " FROM tsdb_group "
-			+ "WHERE group_id = " + groupId;
+			+ "WHERE group_id = ?";
 		try
 		{
-			ResultSet rs = doQuery(q);
-			if (rs != null && rs.next())
+			ret = getSingleResultOr(q, rs -> rs2group(rs), (TsGroup)null, groupId);
```

Comment:
does the parameter argument(s), in this case groupId, need to be a primitive type?   groupId is a DbKey .

Reply:
No, the parameters array is of type Object, the "bind" function in DaoBase takes care of appropriate SQL type mapping.